notebooks/02-Summary-statistics-and-plots.ipynb

```diff
@@ -0,0 +1,518 @@
+    "plt.title(\"Minimum Feret\")\n",
+    "plt.ylabel(\"Minimum Feret / nm\")\n",
+    "# Define text for the string to go in a blue text box.\n",
+    "text_str = \"\\n\".join(\n",
+    "    [\"Sodium Concentration    : 0.001mM\", \"Scan Size                        : 200x200\", \"More useful information : :-)\"]\n",
+    ")\n",
+    "props = dict(boxstyle=\"round\", alpha=0.5)\n",
+    "ax.text(\n",
+    "    0.5,\n",
+    "    0.85,\n",
+    "    text_str,\n",
+    "    transform=ax.transAxes,\n",
+    "    fontsize=12,  # verticalalignment=\"top\",\n",
+    "    horizontalalignment=\"center\",\n",
+    "    bbox=props,\n",
+    ")\n",
+    "# Return the index\n",
+    "df_three_images.set_index([\"image\", \"threshold\", \"molecule_number\"], inplace=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8f56b8fa-8c55-47a5-9d32-d461a1c639b6",
+   "metadata": {},
+   "source": [
+    "### Joint Plot\n",
+    "[Joint  Plots](https://seaborn.pydata.org/generated/seaborn.jointplot.html) showing the relationship between two variables can be plotted easily."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6aa28c57-66fb-4857-b0df-9ac8f2725173",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df.columns\n",
+    "sns.jointplot(data=df, x=\"min_feret\", y=\"max_feret\", hue=\"circular\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "300632da-8be7-4017-a958-f3ebbec01c1d",
+   "metadata": {},
+   "source": [
+    "# Loading Pickles\n",
+    "\n",
+    "One of the features of the `topostats.plotting` module is that it will save the graphs it has generated in Python's own\n",
+    "format for storing objects. These are known as [pickles](https://realpython.com/python-pickle-module/) and typically\n",
+    "have the file extension `.pkl`. After running TopoStats and plotting distributions of statistics a `.pkl` will by\n",
+    "default be saved in the output directory. Here we load  (\"unpickle\") such a saved file which is typically located at\n",
+    "~output/summary_distributions/distribution_plots.pkl~ and check its type (its a ~dict~ionary of plots).\n",
+    "\n",
+    "TopoStats includes the ~load_pkl~ function which loads and returns the contents of the file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fded1124-b8f3-44b8-b73e-331079ebdce0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pathlib import Path\n",
+    "\n",
+    "from topostats.io import load_pkl\n",
+    "\n",
+    "# NB - If you get an error saying the file could not be found then you need to modify this line to point to the location\n",
+    "# the file was saved to.\n",
+    "pkl_path = Path(\"../output/summary_distributions/distribution_plots.pkl\")\n",
+    "my_plots = load_pkl(pkl_path)\n",
+    "# Show the type of my_plots which is a dictionary of nested dictionaries\n",
+    "type(my_plots)"
+   ]
+  },
```

Comment:
Hmmm :thinking: I hadn't thought of that, because I'd already generated a pickle and was using it locally and it didn't click that I needed to perhaps provide one. It adds more steps/configuration to the `package.data` stage to include one or I could instead write something along the lines of...

> If you have already `run_topostats` with the defaults then it will have generated a `output/summary_distributions/distribution_plots.pkl` file which you can load and modify. If you have not done so and wish to modify plots in this manner you will have to execute `run_topostats` from within the target directory before you can load such figures and modify them.

Thoughts on this appreciated, its another situation where we could try and cover every eventuality that might arise but that isn't pragmatic so we have to make a choice and direct people towards a particular course of action. It's not there yet but I'd like the Notebooks to run contiguously such that the results of `00` and/or `01` Notebooks feed into this one (`02`) and it "just works" but that would still be somewhat lacking in explaining how to adapt them to be used on any directory/own use case. :thinking:

Reply:
What if you just have a little step in the notebook to run it (via the % magic if that'll work) or state at the top that a successful run is needed and to see notebook 1 to see how that is done?